query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/condition/ConditionalStatement.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.condition;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilderDsl;
+import com.datastax.oss.driver.api.querybuilder.term.Term;
+import com.datastax.oss.driver.internal.querybuilder.condition.DefaultConditionBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.ColumnComponentLeftHandSide;
+import com.datastax.oss.driver.internal.querybuilder.relation.ColumnLeftHandSide;
+import com.datastax.oss.driver.internal.querybuilder.relation.FieldLeftHandSide;
+import java.util.Arrays;
+
+/**
+ * A statement that can be applied conditionally, such as UPDATE or DELETE.
+ *
+ * <p>Not that this does not covert INSERT... IF NOT EXISTS, which is handled separately.
+ */
+public interface ConditionalStatement<SelfT extends ConditionalStatement<SelfT>> {
+
+  /**
+   * Adds an IF EXISTS condition.
+   *
+   * <p>If any column conditions were added before, they will be cleared.
+   */
+  SelfT ifExists();
+
+  /**
+   * Adds an IF condition. All conditions are logically joined with AND. If {@link #ifExists()} was
+   * invoked on this statement before, it will get cancelled.
+   *
+   * <p>To create the argument, use one of the {@code ifXxx} factory methods in {@link
+   * QueryBuilderDsl}, for example {@link QueryBuilderDsl#ifColumn(CqlIdentifier) isColumn}.
+   *
+   * <p>If you add multiple conditions as once, consider {@link #if_(Iterable)} as a more efficient
+   * alternative.
+   */
+  SelfT if_(Condition condition);
```

Comment:
Not super happy with the name, but it's the best I've found to avoid colliding with the Java keyword. I had `ifCondition` at some point...